src/CoreBundle/Bridge/Chameleon/Module/NavigationTree/NavigationTree.php

```diff
@@ -0,0 +1,740 @@
+        }
+
+        return $name;
+    }
+
+    private function setTypeAndAttributes(BackendTreeNodeDataModel $treeNodeDataModel, \TdbCmsTree $node): void
+    {
+        $treeNodeDataModel->setType($this->setInitialType($node));
+
+        $liAttr = $treeNodeDataModel->getLiAttr();
+        $aAttr = [];
+        $liClass = '';
+        $aClass = '';
+
+        $nodeHidden = false;
+        if (true == $node->fieldHidden) {
+            $nodeHidden = true;
+            $aClass .= ' node-hidden';
+        }
+
+        $pages = [];
+        $connectedPages = $node->GetAllLinkedPages();
+        $foundSecuredPage = false;
+        while ($connectedPage = $connectedPages->Next()) {
+            $pages[] = $connectedPage->id;
+
+            if (false === $nodeHidden
+                && false === $foundSecuredPage
+                && (true === $connectedPage->fieldExtranetPage
+                    && false === $node->fieldShowExtranetPage)) {
+                $aClass .= ' page-hidden';
+                $treeNodeDataModel->setType('page-hidden');
+                $foundSecuredPage = true;
+            }
+
+            if (true === $connectedPage->fieldExtranetPage) {
+                $treeNodeDataModel->setType('locked');
+                $aClass .= ' locked';
+            }
+        }
+
+        if (count($pages) > 0) {
+            if ('folder' === $treeNodeDataModel->getType()) {
+                $treeNodeDataModel->setType('folderWithPage');
+            }
+
+            $primaryPageID = $node->GetLinkedPage(true);
+            if (false !== $primaryPageID) {
+                $liAttr = array_merge($liAttr, ['isPageId' => \TGlobal::OutHTML($primaryPageID)]);
+            }
+
+            // current page is connected to this node
+            if ($this->currentPageId && in_array($this->currentPageId, $pages)) {
+                $aClass .= ' activeConnectedNode';
+                $treeNodeDataModel->setOpened(true);
+                $treeNodeDataModel->setSelected(true);
+
+                if ($this->primaryConnectedNodeIdOfCurrentPage === $node->id) {
+                    $aClass .= ' primaryConnectedNode';
+                    $treeNodeDataModel->setDisabled(true);
+                }
+            } else {
+                $aClass .= ' otherConnectedNode';
+            }
+        }
+
+        $liAttr = array_merge($liAttr, ['class' => $liClass]);
+        $treeNodeDataModel->setLiAttr($liAttr);
+
+        $aAttr = array_merge($aAttr, ['class' => $aClass]);
+        $treeNodeDataModel->setAAttr($aAttr);
+    }
+
+    private function setInitialType(\TdbCmsTree $node): string
+    {
+        $children = $node->GetChildren(true);
+        if ($children->Length() > 0) {
+            $type = 'folder';
+        } else {
+            $type = 'page';
+        }
+
+        if ('' !== $node->sqlData['link']) {
+            $type = 'externalLink';
+        }
+
+        if (in_array($node->id, $this->restrictedNodes)) {
+            $type .= 'RestrictedMenu';
```

Comment:
'RestrictedMenu' is appended in this case. The result is 'folderRestricedMenu' or 'pageRestrictedMenu'
The type 'node-hidden' and 'page-hidden' can be renamed to camelcase, it was like this because the css class is also called that.